src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,601 @@
+    return count;
+}
+
+//------------------------------------------------------------------------------
+// TiltActuator
+//------------------------------------------------------------------------------
+
+void WindowCover::TiltActuator::OpenLimitSet(uint16_t limit)
+{
+    WindowCovering::Attributes::SetInstalledOpenLimitTilt(mEndPoint, limit);
+}
+
+uint16_t WindowCover::TiltActuator::OpenLimitGet(void)
+{
+    uint16_t limit = 0;
+    WindowCovering::Attributes::GetInstalledOpenLimitTilt(mEndPoint, &limit);
+    return limit;
+}
+
+void WindowCover::TiltActuator::ClosedLimitSet(uint16_t limit)
+{
+    WindowCovering::Attributes::SetInstalledClosedLimitTilt(mEndPoint, limit);
+}
+
+uint16_t WindowCover::TiltActuator::ClosedLimitGet(void)
+{
+    uint16_t limit = 0;
+    WindowCovering::Attributes::GetInstalledClosedLimitTilt(mEndPoint, &limit);
+    return limit;
+}
+
+void WindowCover::TiltActuator::PositionSet(uint16_t percent100ths)
+{
+    uint8_t percent = static_cast<uint8_t>(percent100ths / 100);
+    uint16_t value  = Percent100thsToValue(percent100ths);
+
+    WindowCovering::Attributes::SetCurrentPositionTilt(mEndPoint, value);
+    WindowCovering::Attributes::SetCurrentPositionTiltPercentage(mEndPoint, percent);
+    WindowCovering::Attributes::SetCurrentPositionTiltPercent100ths(mEndPoint, percent100ths);
+    emberAfWindowCoveringClusterPrint("TiltActuator::PositionSet(%u%%)", percent100ths / 100);
+}
+
+uint16_t WindowCover::TiltActuator::PositionGet()
+{
+    uint16_t percent100ths = 0;
+    WindowCovering::Attributes::GetCurrentPositionTiltPercent100ths(mEndPoint, &percent100ths);
+    return percent100ths;
+}
+
+void WindowCover::TiltActuator::TargetSet(uint16_t percent100ths)
+{
+    emberAfWindowCoveringClusterPrint("TiltActuator::TargetSet(%u%%)", percent100ths / 100);
+    WindowCovering::Attributes::SetTargetPositionTiltPercent100ths(mEndPoint, percent100ths);
+}
+
+uint16_t WindowCover::TiltActuator::TargetGet()
+{
+    uint16_t percent100ths = 0;
+    WindowCovering::Attributes::GetTargetPositionTiltPercent100ths(mEndPoint, &percent100ths);
+    return percent100ths;
+}
+
+void WindowCover::TiltActuator::NumberOfActuationsIncrement()
+{
+    uint16_t count = static_cast<uint16_t>(NumberOfActuationsGet() + 1);
+    WindowCovering::Attributes::SetNumberOfActuationsTilt(mEndPoint, count);
+}
+
+uint16_t WindowCover::TiltActuator::NumberOfActuationsGet(void)
+{
+    uint16_t count = 0;
+    WindowCovering::Attributes::GetNumberOfActuationsTilt(mEndPoint, &count);
+    return count;
+}
+
+//------------------------------------------------------------------------------
+// Callbacks
+//------------------------------------------------------------------------------
+
+/** @brief Window Covering Cluster Init
+ *
+ * Cluster Init
+ *
+ * @param endpoint    Endpoint that is being initialized
+ */
+void emberAfWindowCoveringClusterInitCallback(chip::EndpointId endpoint)
```

Comment:
Might be better to `using namespace chip;` at the top of the file and not have the `chip::` on everything here.

Reply:
With so many namespaces, I prefer to keep the namespace prefix whenever is short, and does not hinder readability. The prefix provides information of the context to which the type belongs.

Reply:
I can understand that point of view, but there is also value in uniformity of code across the codebase and the "global" types like EndpointId we generally use without a prefix...  I suspect what will happen is a followup where someone ends up cleaning this up and removing the prefixes anyway....

Anyway, this is not the thing that's blocking the approval here.